src/dataflow/src/render/sources.rs

```diff
@@ -536,13 +539,73 @@ where
                                 SourceEnvelope::None(key_envelope) => {
                                     let results = append_metadata_to_value(results);
 
-                                    let (stream, errors) =
-                                        flatten_results_prepend_keys(key_envelope, results)
-                                            .ok_err(ResultExt::err_into);
-                                    let stream = stream.pass_through("decode-ok").as_collection();
-                                    let errors =
-                                        errors.pass_through("decode-errors").as_collection();
-                                    (stream, Some(errors))
+                                    let flattened_stream =
+                                        flatten_results_prepend_keys(key_envelope, results);
+
+                                    // When persistence is enabled we need to persist and seal up
+                                    // both the timestamp bindings and the data. Otherwise, just
+                                    // pass through.
+                                    let (flattened_stream, persist_errs) = if let Some(
+                                        source_persist_config,
+                                    ) =
+                                        source_persist_config
+                                    {
+                                        let bindings_config =
+                                            &source_persist_config.bindings_config;
+                                        let envelope_config =
+                                            source_persist_config.envelope_none_config();
+
+                                        let (flattened_stream, persist_errs) =
+                                            super::envelope_none::persist_and_replay(
```

Comment:
nit: add a use for envelope_none at the top of the file

Reply:
:+1: